hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java

```diff
@@ -419,31 +427,32 @@ public synchronized void addPersistedDelegationToken(
     }
   }
 
-  /** 
-   * Update the current master key 
-   * This is called once by startThreads before tokenRemoverThread is created, 
+  /**
+   * Update the current master key
+   * This is called once by startThreads before tokenRemoverThread is created,
    * and only by tokenRemoverThread afterwards.
    */
-  private void updateCurrentKey() throws IOException {
+  protected void updateCurrentKey() throws IOException {
     LOG.info("Updating the current master key for generating delegation tokens");
     /* Create a new currentKey with an estimated expiry date. */
     int newCurrentId;
     synchronized (this) {
-      newCurrentId = incrementCurrentKeyId();
+      newCurrentId = generateNewKeyId();
```

Comment:
Why should we change the method name？

Reply:
the actual requirement is to generate new keys and not to increment keys - this greatly simplifies implementation for various DB systems

Reply:
We use an incremental method to generate new keys. For the database, we use for...update to ensure the increment. For zk, we use the global generator.

Reply:
yes but there is no requirement to only increment keys 1 by 1. we only require to generate new unique keys. hence the name change to clearly reflect that requirement.

Reply:
> yes but there is no requirement to only increment keys 1 by 1. we only require to generate new unique keys. hence the name change to clearly reflect that requirement.

The performance of the database (Mysql) should be better, increase one by one, the performance should be ok.

ZK We are applying for multiple ids at one time.

Users can configure the parameter `zk-dt-secret-manager.token.seqnum.batch.size` to apply for multiple `sequenceNum` at a time.

ZookeeperFederationStateStore#incrSharedCount
```
private int incrSharedCount(SharedCount sharedCount, int batchSize)
      throws Exception {
    while (true) {
      // Loop until we successfully increment the counter
      VersionedValue<Integer> versionedValue = sharedCount.getVersionedValue();
      if (sharedCount.trySetCount(versionedValue, versionedValue.getValue() + batchSize)) {
        return versionedValue.getValue();
      }
    }
  }
```